src/Chapter3.hs

```diff
@@ -1006,6 +1161,79 @@ Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
 
+data MonsterActions = MAttack | Run
+    deriving (Show, Eq, Enum, Bounded)
+
+data Monster' = M 
+    { mHealth  :: Health
+    , mAttack  :: Attack
+    , mActions :: MonsterActions
+    } deriving Show
+
+data KnightActions 
+    = KAttack
+    | Heal
+    | SpellCast
+    deriving (Show, Eq, Enum, Bounded)
+  
+data Knight' = K
+    { kHealth  :: Health
+    , kAttack  :: Attack
+    , kDefense :: Defense
+    , kActions :: KnightActions
+    } deriving Show
+
+cycleMActions :: MonsterActions -> MonsterActions
+cycleMActions a
+    | a == maxBound   = minBound
+    | otherwise       = succ a
```

Comment:
Interesting solution :thinking: It's correct, and we haven't required anything fancy here. Though, if you want to allow different sequences of actions for different values of the `Monster` type, you will need to model this differently :slightly_smiling_face: